s3fs/core.py

```diff
@@ -894,6 +894,13 @@ def copy_basic(self, path1, path2, **kwargs):
             raise ValueError('Copy failed (%r -> %r): %s' % (path1, path2, e))
         self.invalidate_cache(path2)
 
+
+    @staticmethod
+    def _get_brange(size, block):
```

Comment:
Sorry, what is this?
Since it's a function, I would rather it didn't clutter the spec namespace. It could be in utils, ideally with a leading underscore.

Reply:
Agreed, I've moved it. But still need to figure out why `test_copy_managed` passes before and after my change to `copy_managed`.

Reply:
Have you tried with totally random data?

Reply:
In `test_copy_managed` I just replaced:

```data = b'abc' * 12*2**20```
with
```data = os.urandom(3*12*2**20)```

And, as of yet, I still can't get the test to fail (on master).

For reference here are the two byte ranges (the only difference being the very last digit).

Master branch 
```
37748736
bytes=0-5242879
bytes=5242880-10485759
bytes=10485760-15728639
bytes=15728640-20971519
bytes=20971520-26214399
bytes=26214400-31457279
bytes=31457280-36700159
bytes=36700160-37748736
```

This branch
```
37748736
bytes=0-5242879
bytes=5242880-10485759
bytes=10485760-15728639
bytes=15728640-20971519
bytes=20971520-26214399
bytes=26214400-31457279
bytes=31457280-36700159
bytes=36700160-37748735
```

Reply:
and if you run this against real S3, the master fails but this branch passes?

Reply:
That's right.  The master gives:
 
```OSError: Write failed: ClientError('An error occurred (InvalidArgument) when calling the UploadPartCopy operation: Range specified is not valid for source object of size: 6403334120')```

The branch completes the copy without error (although I have not verified the content of the new file).

Reply:
Also, could you please check the difference when the total data does/doesn't divide exactly by the block size?